MetaMask/metamask-extension
# feat: Add metrics for simulation warnings.

## **Description**

1. Add a new boolean property to Transactions events named gas_estimation_failed and pass true value to it whenever simulationFails property is added to the txMeta
2. Append gas_estimation_failed within ui_customizations property of Transactions events whenever we display gas estimation failed warning.

This will help us identify in which transactions we're failing to estimate gas and how the user is behaving. We don't need to specifically track clicking on the I will try anyways as we initially can defer that from the user submiting the transaction afterwards.


## **Manual testing steps**
1. Start extension 
2. Run a contract method that will fail, so that the simulationError will be shown (you can alternatively edit the code to [remove the condition here](https://github.com/MetaMask/metamask-extension/blob/d6d324bd2ac1fcf760e438329bf958735a932f1f/ui/components/app/transaction-alerts/transaction-alerts.js#L104-L105) to show the error always)
3. Check metrics to see the above properties are added to Transaction events and ui_customization has the correct property

## **Related issues**

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
